### ISSUE TITLE
Exception causes get lost when using IExecutorService with the hazelcast client

### ISSUE THREAD

**Issue by Petikoch** [hazelcast/hazelcast#10591]
In a distributed java application, I switched over in a node from a regular java hazelcast light member to a java hazelcast client. I noticed, that the hazelcast client behaves slightly different in terms of transportation of exceptions in the IExecutorService: exception causes get **lost**.

In the following example (using 3.7.7), we have three exceptions in a chain

- RuntimeException "a"
- with its cause IllegalStateException "b"
- with its cause IllegalArgumentException "c"

Using the hazelcast client we see

- RuntimeException "a" is transported
- cause IllegalStateException is transported, but message "b" is missing
- the actual root cause IllegalArgumentException "c" is missing completely

```
import com.hazelcast.client.HazelcastClient;
import com.hazelcast.core.Hazelcast;
import com.hazelcast.core.HazelcastInstance;
import org.hamcrest.Matchers;

import java.io.Serializable;
import java.util.Objects;
import java.util.concurrent.Callable;
import java.util.concurrent.ExecutionException;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertThat;

public class ExceptionCausesLost {

	public static void main(String[] args) throws InterruptedException {
		HazelcastInstance hzServer = Hazelcast.newHazelcastInstance();
		hzServer.getCluster().getLocalMember().setStringAttribute("type", "server");

		// exception causes get lost with client:
		HazelcastInstance hzClient = HazelcastClient.newHazelcastClient();
		// works, in contrast:
		//HazelcastInstance hzClient = Hazelcast.newHazelcastInstance();

		try {
			hzClient.getExecutorService("Default").submit(
					new ThrowingCallable(),
					member -> Objects.equals("server", member.getStringAttribute("type"))
			).get();
			throw new IllegalStateException("Should not be here, should be in catch ExecutionException block");
		} catch (ExecutionException e) {
			assertThat(e.getCause(), Matchers.instanceOf(RuntimeException.class));
			assertEquals("a", e.getCause().getMessage());
			assertThat(e.getCause().getCause(), Matchers.instanceOf(IllegalStateException.class));
			assertEquals("b", e.getCause().getCause().getMessage());
			assertThat(e.getCause().getCause().getCause(), Matchers.instanceOf(IllegalArgumentException.class));
			assertEquals("c", e.getCause().getCause().getCause().getMessage());
			System.out.println("Everything is fine...");
		} finally {
			hzClient.shutdown();
			hzServer.shutdown();
		}
	}

	public static class ThrowingCallable implements Callable<String>, Serializable {

		@Override
		public String call() throws Exception {
			throw new RuntimeException("a", new IllegalStateException("b", new IllegalArgumentException("c")));
		}

	}
}
```

Expected: 
The full exception chain is transported with the java hazelcast client, like with "regular" hazelcast nodes.

**Comment by sancar** [comment]
fixed via https://github.com/hazelcast/hazelcast/pull/9750 in 3.8

**Comment by Petikoch** [comment]
@sancar this is fixed in 3.9 not 3.8, isn't it?

**Comment by sancar** [comment]
I just checked the fix commit.
https://github.com/hazelcast/hazelcast/pull/9750/commits/e9a783c1022425fe3fe27ef7fff24bb1d4f308a5
It is shipped with 3.8. Label of this issue was not properly updated. I've just changed it to 3.8

**Comment by Petikoch** [comment]
Ok, I saw it in the release notes of 3.9... This explains it.

Thanks a lot and have a nice day!
